### PR TITLE
manual_control: add topic for sending MANUAL_CONTROL messages

### PR DESCRIPTION
**Pull request by okalachev**
PX4 supports "overriding" manual control inputs via MANUAL_CONTROL message: https://github.com/PX4/Firmware/blob/master/src/modules/mavlink/mavlink_receiver.cpp#L1559 (did not check for APM).

I added `~manual_control/override` topic (not sure about the name, this is the analogy with `rc/override`), allowing user to control the copter with a "virtual" RC with normalised values. For example:

```
rostopic pub -r 5 /mavros/manual_control/override mavros_msgs/ManualControl "header:
  seq: 0
  stamp: {secs: 0, nsecs: 0}
  frame_id: ''
x: 0.0
y: 0.0
z: 1000.0
r: 200.0
buttons: 0" 
```

in `POSCTL` mode makes the copter fly up and rotate (tested in SITL).

Also, I think `1` is the correct value for `queue_size` for such topics (`rc/override`, `manual_control/override`, etc), because we want to send relevant messages, we don't want to queue them.

### CODE REVIEW

File: mavros/src/plugins/manual_control.cpp [hunk 1]
```diff
@@ -34,6 +34,7 @@ class ManualControlPlugin : public plugin::PluginBase {
 		PluginBase::initialize(uas_);
 
 		control_pub = manual_control_nh.advertise<mavros_msgs::ManualControl>("control", 10);
+		control_sub = manual_control_nh.subscribe("override", 1, &ManualControlPlugin::override_cb, this);
```
Comment (vooon) [inline mavros/src/plugins/manual_control.cpp]: Should be called `override_sub` in case topic name `override`. Maybe name it `send`?

Comment (okalachev) [reply mavros/src/plugins/manual_control.cpp]: Yes, `send` is better. But, in such case, there'll be two topics, `send` and `control`, it's non-obvious, that these are send and receive, actually.

**Comment by okalachev** [comment]
I renamed `override` to `send`. I think, the best naming is something like `send` and `receive`, but this will break up the compatibility, which is not good.

**Comment by mchen0037** [comment]
@okalachev Could you give a brief explanation of how this works? When I use the example you give 
```
rostopic pub -r 5 /mavros/manual_control/override mavros_msgs/ManualControl "header:
  seq: 0
  stamp: {secs: 0, nsecs: 0}
  frame_id: ''
x: 0.0
y: 0.0
z: 1000.0
r: 200.0
buttons: 0" 
```
My copter flies straight up while spinning about yaw , but if I pass 0 in for z, it just sits there.

Is there a way to pass in normalized RC values? 

Thanks!

**Comment by okalachev** [comment]
First, the topic was renamed to `send` (`/mavros/manual_control/send`). So it should not work at all :)

**Comment by mchen0037** [comment]
Sorry--I just copy-pasted. I'm publishing it to /mavros/manual_control/send

**Comment by okalachev** [comment]
In PX4, `x` and `y` are the pitch and roll sticks, normalized from -1000 to 1000.
`z` is a throttle stick, normilized from 0 to 1000 (unlike in MAVLink documentation).
`r` is yaw stick, normilized from -1000 to 1000.

Don't you have it working this way?

**Comment by mchen0037** [comment]
Okay, thanks! I was sending in yaw values into z and throttle into r.

**Comment by romaster93** [comment]
@okalachev 
Is there any further explanation possible?

I modified the command to 'send'. It is output normally in rostopic ehco manual_control / send. However, the FCU is not responding.

**Comment by okalachev** [comment]
Sorry, I don't get your question. If publish values to this topic than the vehicle should behave as it was a real RC.

**Comment by romaster93** [comment]
ok. 
I will attach a picture

**Comment by romaster93** [comment]
@okalachev 

my env : 
px4 firmware
gazebo 9
ubuntu 18.04
ros melodic

i tyed this :

roslaunch mavros px4.launch

make px4_sitl_default gazebo

./QGroundControl.AppImage

rosrun test mavkbteleop.py ( https://github.com/romaster93/px4_test/blob/master/mavkbteleop.py )

and All work.


I can takeoff using QGroundControl. 

and i checked ' rostopic echo /mavros/rc/out ' :

header :

         seq: 225

         stamp:

             secs: 1587478017

             nsecs: 334894783

         frame_id : ' '

channels : [900, 900, 900, 900, 900, 900, 0 ,0, 0 ,0 ,0 ,0 ,0 ,0 ,0]

 

check 'rostopic echo /mavros/rc/in : 

header: 
  seq: 194
  stamp: 
    secs: 1587478017
    nsecs: 291170812
  frame_id: ''
rssi: 100
channels: [1500, 1500, 1500, 1500, 1000, 1000, 1000, 1000, 61566, 21897, 0, 46240, 43647, 32532, 0, 15184, 37888, 32532]

check 'rostopic echo /mavros/rc/override' :

channels: [1500, 1500, 1500, 1500, 1000, 1000, 1000, 1000]


first, I believed that the drone would respond if I modified the override value.
but, the drone did not move. So while I was looking for a solution, I saw an article In google that responded by tying terminal the ' rosrun mavros mavparm set SYSID_MYGCS 1 ', but it didn't work for me. In px4, SYSID_MYGCS did not exist.

After, I saw your article and tried it. I typed in another terminal.

rostopic pub -r 5 /mavros/manual_control/send mavros_msgs/ManualControl "header:
  seq: 0
  stamp: {secs: 0, nsecs: 0}
  frame_id: ''
x: 0.0
y: 0.0
z: 1000.0
r: 200.0
buttons: 0" 

I entered in terminal to confirm that the message was delivered.

rostopic echo mavros/manual_control/send :

header: 
  seq: 1675
  stamp: 
    secs: 0
    nsecs:         0
  frame_id: ''
x: 0.0
y: 0.0
z: 1000.0
r: 200.0
buttons: 0
---

But my drone(in gazebo) did not move on the ground.

I definitely want to launch the drone through the rc value in gazebo. I want to know what my problem is.

thanks....

**Comment by okalachev** [comment]
Basically you have to arm your drone before it will handle virtual RC commands. Maybe this is the problem?

Also, if you want you drone to fly autonomously, it's generally better to utilize the OFFBOARD mode and setpoint commands.

**Pull request merged**